appium/options/android/common/adb/adb_options.py

```diff
@@ -0,0 +1,227 @@
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from datetime import timedelta
+from typing import Any, TypeVar
+
+from appium.options.common.supports_capabilities import SupportsCapabilities
+
+C = TypeVar("C", bound="SupportsCapabilities")
+
+
+class AdbOptionsDescriptor:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def __get__(self, obj: C, cls: type[C]) -> Any:
+        if self.name == 'ADB_EXEC_TIMEOUT':
+            value = getattr(obj, 'get_capability')(self.name)
+            return None if value is None else timedelta(milliseconds=value)
+        return getattr(obj, 'get_capability')(self.name)
+
+    def __set__(self, obj:C, value: C) -> C:
+        if self.name == 'ADB_EXEC_TIMEOUT':
+            return getattr(obj, 'set_capability')(self.name, int(value.total_seconds() * 1000) if isinstance(value, timedelta) else value)    
+        return getattr(obj, 'set_capability')(self.name, value)
+
+
+class AdbExecTimeoutOption(SupportsCapabilities):
+    ADB_EXEC_TIMEOUT = 'adbExecTimeout'
+    adb_exec_timeout = AdbOptionsDescriptor("ADB_EXEC_TIMEOUT")
+    """
```

Comment:
this docustring is not very helpful as it is not going to be shown by IDE for the corresponding property

Reply:
> this docustring is not very helpful as it is not going to be shown by IDE for the corresponding property

The docstring shows up in IDE. I am using VSCode. Attaching screenshot below.

<img width="837" alt="docstrings" src="https://github.com/appium/python-client/assets/26169602/b2b97351-94e5-4120-9a5f-3501e775814a">

Reply:
it looks ok, although is still not ideal as there is no proper typehint for the value, so we don't actually know what valid objects could be assigned to it and what the returned type is

Reply:
> it looks ok, although is still not ideal as there is no proper typehint for the value, so we don't actually know what valid objects could be assigned to it and what the returned type is

We can add notes and have complete information about the type of object that can be assigned and return value Information. I will do this change.

Reply:
> it looks ok, although is still not ideal as there is no proper typehint for the value, so we don't actually know what valid objects could be assigned to it and what the returned type is

The below docstring should be good enough. Let me know your comments/thoughts.

<img width="850" alt="docstring" src="https://github.com/appium/python-client/assets/26169602/5a94fc70-78a9-4ddc-b461-ceecd00d7f7a">

Reply:
this docstring looks better to me

Reply:
> this docstring looks better to me

Thank You for your feedback. I will apply the same style to all the other descriptors.

Reply:
> this docstring looks better to me

This looks even better and more informative about each `gettter` and `setter`.

<img width="998" alt="doc_strings" src="https://github.com/appium/python-client/assets/26169602/aedb4dff-0265-4d8a-ae1d-09bcc7ef55c9">